homeassistant/components/husqvarna_automower/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow to add the integration via the UI."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import data_entry_flow
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HusqvarnaConfigFlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler,
+    data_entry_flow.FlowHandler,
+    domain=DOMAIN,
+):
+    """Handle a config flow."""
+
+    DOMAIN = DOMAIN
+    VERSION = 1
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create an entry for the flow."""
+        return await self.async_step_finish(DOMAIN, data)
+
+    async def async_step_finish(self, unique_id, data) -> FlowResult:
+        """Complete the config entries."""
+        await self.async_set_unique_id(unique_id)
+
+        return self.async_create_entry(
+            title=unique_id,
+            data=data,
+        )
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            _LOGGER.debug("USER INPUT NONE")
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({}),
+            )
+        return await self.async_step_user()
```

Comment:
If possible, let's move reauth to a follow up PR so we can focus on the rest and make it as small as possible. The reauth still might need some work when looking at logging in with the right account (for example, you create the config with account A, and reauth with acount B). But let's keep that out of scope for now, and instead of ConfigEntryAuthFailed, just raise ConfigEntryNotReady for the time being